examples/json.rs

```diff
@@ -17,24 +17,25 @@ enum Json {
     Object(HashMap<String, Json>),
 }
 
-fn parser() -> impl Parser<char, Json, Error = Simple<char>> {
+fn parser<'a>() -> impl Parser<'a, str, Json, Rich<str>, ()> {
     recursive(|value| {
-        let frac = just('.').chain(text::digits(10));
+        let frac = just('.').then_ignore(text::digits(10))
+            .map_slice(|x| x);
```

Comment:
Is this a common enough operation that we could do with a `.slice()` combinator?

Reply:
Probably, basically all the non-trivial examples use it at least once or twice (including nano_rust's non-pushed WIP port I have in my local, I'm still considering how best to port its spans)